test/auth.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { auth as native, http as native_http } from '../lib/index';
+import { AwsCredentialsProvider } from '../lib/native/auth';
+
+test('AwsSigningConfig properties', () => {
+    const now = new Date();
+    const provider = AwsCredentialsProvider.newStatic("key", "secret");
+    let config = new native.AwsSigningConfig(
+        native.AwsSigningAlgorithm.SigV4QueryParam,
+        provider,
+        'us-east-1',
+        'iotcore',
+        now,
+        ["abc"],
+        true,
+        true,
+        true,
+    );
+
+    expect(config.algorithm).toBe(native.AwsSigningAlgorithm.SigV4QueryParam);
+
+    expect(config.region).toBe('us-east-1');
+
+    expect(config.service).toBe('iotcore');
+
+    expect(config.date).toBe(now);
+
+    expect(config.param_blacklist.length).toBe(1);
+    expect(config.param_blacklist).toContain("abc");
+    config.param_blacklist.push("def");
+    expect(config.param_blacklist.length).toBe(2);
+    expect(config.param_blacklist).toContain("abc");
+    expect(config.param_blacklist).toContain("def");
+
+    expect(config.use_double_uri_encode).toBe(true);
+
+    expect(config.should_normalize_uri_path).toBe(true);
+
+    expect(config.sign_body).toBe(true);
+});
+
+const DATE_STR = '2015-08-30T12:36:00Z';
+
+// Test values copied from aws-c-auth/tests/aws-sig-v4-test-suite/get-vanilla"
```

Comment:
this comment isn't accurate anymore.
I have no idea where you got this Authorization signature from. I searched for the string in aws-c-auth and don't see where it came from.

Reply:
Ah!
I got the python test results matching those from aws-c-auth!

You need to set sign_body=false.
The "x-amz-content-sha256" header will still get added, but it won't be accounted for in the "Authorization" signature.  I made the python test less strict, checking that the signed headers were PRESENT, but not don't fail if there are extra headers present